### ISSUE TITLE
[Accessibility]: Global nav bar: User profile menu usability

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35119]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34411

### Problem description

**Focus styles**

<img width="323" alt="image" src="https://user-images.githubusercontent.com/9516420/167417449-864f93a5-aab1-4b9e-b703-103e92adc760.png">

There isn't a clear blue outline focus when the menu has keyboard focus. It should appear like other elements do, such as the navigation links.

**Menu items**

When using a screen reader, links only appear as "MenuItem" - they should also read out the href and any other link related information to the user.

### Expected behavior

**Focus styles**

Add clear blue focus outline when selected

**Menu items**
Ensure screen readers still treat each item as links. Note: we still want a menu item role somewhere, to support keyboard navigation. Maybe we can wrap each menu item in the link.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
Might be fixed by https://github.com/sourcegraph/sourcegraph/issues/36350
